plugins/aea-cli-ipfs/aea_cli_ipfs/core.py

```diff
@@ -86,9 +91,27 @@ def add(
     """Add directory to ipfs, if not directory specified the current one will be added."""
     dir_path = dir_path or os.getcwd()
     ipfs_tool = click_context.obj
+
     click.echo(f"Starting processing: {dir_path}")
     name, hash_, _ = ipfs_tool.add(dir_path, pin=(not no_pin))
-    click.echo(f"Added: `{name}`, hash is {hash_}")
+
+    yaml_files = glob(str(Path(dir_path) / "*.yaml"))
+    if len(yaml_files) > 0:
+        (config_file_path,) = yaml_files
+        package_path, config_file = os.path.split(config_file_path)
+        item_type, _ = os.path.splitext(config_file)
+        package_path = Path(package_path)
+        item_type = "agent" if item_type == "aea-config" else item_type
+        item_config = load_item_config(item_type=item_type, package_path=package_path)
+        register_item_to_local_registry(
+            item_type=item_type, public_id=item_config.public_id, package_hash=hash_,
+        )
+        click.echo(
+            f"Registered item with:\n\tpublic id : {item_config.public_id}\n\thash : {hash_}"
+        )
```

Comment:
The package hash won't be stored inside component.yaml now, Instead it'll be stored in a local registry file located at `~/.aea/local_registry.json`.